addons/account/models/chart_template.py

```diff
@@ -361,7 +361,11 @@ def deref(values, model):
                     field.type == 'many2one'
                     or (field.type in ('integer', 'many2one_reference') and not value.isdigit())
                 ):
-                    values[fname] = self.ref(value).id if value not in ('', 'False', 'None') else False
+                    try:
+                        values[fname] = self.ref(value).id if value not in ('', 'False', 'None') else False
+                    except ValueError as e:
+                        _logger.warning("Failed when trying to recover %s for field=%s", value, field)
+                        raise e
```

Comment:
For example, let's say I put a wrong xml_id for a tax_group_id ref in `account.tax-be.csv`:
![image](https://github.com/odoo/odoo/assets/52834046/7820b9c8-eb11-45f8-85c5-2f44bb94b205)

And that, I run the test class `TestOSSBelgium` then I get this in the console:
```
2023-05-31 09:21:19,825 18413 WARNING test odoo.addons.account.models.chart_template: Failed when trying to recover bob_moisi for field=account.tax.tax_group_id 
2023-05-31 09:21:19,827 18413 INFO test odoo.tests.suite: ====================================================================== 
2023-05-31 09:21:19,827 18413 ERROR test odoo.tests.suite: ERROR: setUpClass (odoo.addons.l10n_eu_oss.tests.test_oss.TestOSSBelgium)
Traceback (most recent call last):
  File "/home/odoo/src/odoo/odoo/tools/cache.py", line 91, in lookup
    r = d[key]
  File "<decorator-gen-3>", line 2, in __getitem__
  File "/home/odoo/src/odoo/odoo/tools/func.py", line 87, in locked
    return func(inst, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('ir.model.data', <function IrModelData._xmlid_lookup at 0x7f72d2d45e50>, 'account.4_bob_moisi')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/odoo/src/odoo/addons/l10n_eu_oss/tests/test_oss.py", line 26, in setUpClass
    cls.load_specific_chart_template(chart_template_ref)
  File "/home/odoo/src/odoo/addons/l10n_eu_oss/tests/test_oss.py", line 19, in load_specific_chart_template
    raise e
  File "/home/odoo/src/odoo/addons/l10n_eu_oss/tests/test_oss.py", line 14, in load_specific_chart_template
    super().setUpClass(chart_template_ref=chart_template_ref)
  File "/home/odoo/src/odoo/addons/account/tests/common.py", line 65, in setUpClass
    cls.company_data_2 = cls.setup_company_data('company_2_data', chart_template=chart_template_ref)
  File "/home/odoo/src/odoo/addons/account/tests/common.py", line 205, in setup_company_data
    cls.env['account.chart.template'].try_loading(chart_template, company=company, install_demo=False)
  File "/home/odoo/src/odoo/addons/account/models/chart_template.py", line 141, in try_loading
    return self._load(template_code, company, install_demo)
  File "/home/odoo/src/odoo/addons/account/models/chart_template.py", line 185, in _load
    self._load_data(data)
  File "/home/odoo/src/odoo/addons/account/models/chart_template.py", line 461, in _load_data
    'values': deref(record, self.env[model]),
  File "/home/odoo/src/odoo/addons/account/models/chart_template.py", line 384, in deref
    raise e
  File "/home/odoo/src/odoo/addons/account/models/chart_template.py", line 381, in deref
    values[fname] = self.ref(value).id if value not in ('', 'False', 'None') else False
  File "/home/odoo/src/odoo/addons/account/models/chart_template.py", line 858, in ref
    return self.env.ref(f"account.{self.env.company.id}_{xmlid}" if xmlid and '.' not in xmlid else xmlid, raise_if_not_found)
  File "/home/odoo/src/odoo/odoo/api.py", line 566, in ref
    res_model, res_id = self['ir.model.data']._xmlid_to_res_model_res_id(
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_model.py", line 1988, in _xmlid_to_res_model_res_id
    return self._xmlid_lookup(xmlid)[1:3]
  File "<decorator-gen-40>", line 2, in _xmlid_lookup
  File "/home/odoo/src/odoo/odoo/tools/cache.py", line 96, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/home/odoo/src/odoo/odoo/addons/base/models/ir_model.py", line 1981, in _xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % xmlid)
ValueError: External ID not found in the system: account.4_bob_moisi
```

Reply:
In this case, it's kind of stupid because I can find `bob_moisi` easily but in case of a dynamic formed xml_id, it's less obvious.
This line of log let you, at least, know which model it concerned. (first line in the logs)

Reply:
Be more specific :grinning:

Reply:
meh.